website/docs/r/monitor_data_collection_endpoint.html.markdown

```diff
@@ -0,0 +1,78 @@
+---
+subcategory: "Monitor"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_monitor_data_collection_endpoint"
+description: |-
+  Manages a Data Collection Endpoint.
+---
+
+# azurerm_monitor_data_collection_endpoint
+
+Manages a Data Collection Endpoint.
+
+## Example Usage
+
+```hcl
+# Generated from AccTest TestAccMonitorDataCollectionEndpoint_complete
+provider "azurerm" {
+  features {}
+}
+resource "azurerm_resource_group" "example" {
+  name     = "example-rg"
+  location = "West Europe"
+}
+resource "azurerm_monitor_data_collection_endpoint" "example" {
+  name                          = "example-mdce"
+  resource_group_name           = azurerm_resource_group.example.name
+  location                      = azurerm_resource_group.example.location
+  kind                          = "Windows"
+  public_network_access_enabled = true
+  description                   = "monitor_data_collection_endpoint example"
+  tags = {
+    foo = "bar"
+  }
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `location` - (Required) The Azure Region where the Data Collection Endpoint should exist. Changing this forces a new Data Collection Endpoint to be created.
+
+* `name` - (Required) The name which should be used for this Data Collection Endpoint. Changing this forces a new Data Collection Endpoint to be created.
+
+* `resource_group_name` - (Required) The name of the Resource Group where the Data Collection Endpoint should exist. Changing this forces a new Data Collection Endpoint to be created.
+
+---
+
+* `description` - (Optional) Specifies a description for the Data Collection Endpoint.
+
+* `kind` - (Optional) The kind of the Data Collection Endpoint. Possible values are `Linux` and `Windows`.
```

Comment:
would this make more sense as
```suggestion
* `os_type` - (Optional) The kind of the Data Collection Endpoint. Possible values are `Linux` and `Windows`.
```
?

Reply:
Yes, change `kind` to `os_type` make more sense, but there is a releated resource `data_collection_rule` with the same property `kind` has been merged, so maybe it would be better to keep it consistent.

https://github.com/hashicorp/terraform-provider-azurerm/pull/17342/files#diff-5c76d4d0e7f6a55171a0154b71598e20679ba84a61becbe9b40dee652e5aaa15R126-R128

Reply:
sadly that is probably for the best, maybe we can fix both in 4.0